### ISSUE TITLE
sprite mixin is not compatible with dart-sass

### ISSUE THREAD

**Issue by ChristianStornowski** [cascornelissen/svg-spritemap-webpack-plugin#65]
Changing webpack.config.js in examples variables for dart-sass and running `../../node_modules/.bin/webpack` will result into an error.

### Error:

```bash
ERROR in ./src/scss/style.scss
Module build failed (from /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass-loader/lib/loader.js):

            $sprite: sprite-str-replace($sprite, '___#{$key}___', sprite-svg-uri(quote($value)));
                                                                                ^
      $string: #f00 is not a string.
   ╷
65 │             $sprite: sprite-str-replace($sprite, '___#{$key}___', sprite-svg-uri(quote($value)));
   │                                                                                  ^^^^^^^^^^^^^
   ╵
  src/scss/_sprites.scss 65:82  sprite()
  stdin 4:5                     root stylesheet
      in /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/examples/variables/src/scss/_sprites.scss (line 65, column 82)
    at runLoaders (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/webpack/lib/NormalModule.js:301:20)
    at /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/loader-runner/lib/LoaderRunner.js:364:11
    at /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/loader-runner/lib/LoaderRunner.js:230:18
    at context.callback (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at render (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass-loader/lib/loader.js:52:13)
    at Function.$2 (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:24106:48)
    at w4.$0 (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:15152:8)
    at Object.Gd (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:469:30)
    at Object.hp (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:2216:29)
    at Ha (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:3044:27)
    at /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass/sass.dart.js:3033:42
 @ ./src/index.js 1:0-28
Child mini-css-extract-plugin ../../node_modules/css-loader/dist/cjs.js!../../node_modules/sass-loader/lib/loader.js??ref--4-2!src/scss/style.scss:
    Entrypoint mini-css-extract-plugin = *
    [0] /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/css-loader/dist/cjs.js!/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass-loader/lib/loader.js??ref--4-2!./src/scss/style.scss 803 bytes {0} [built] [failed] [1 error]
    
    ERROR in ./src/scss/style.scss (/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/css-loader/dist/cjs.js!/home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass-loader/lib/loader.js??ref--4-2!./src/scss/style.scss)
    Module build failed (from /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/node_modules/sass-loader/lib/loader.js):
    
                $sprite: sprite-str-replace($sprite, '___#{$key}___', sprite-svg-uri(quote($value)));
                                                                                    ^
          $string: #f00 is not a string.
       ╷
    65 │             $sprite: sprite-str-replace($sprite, '___#{$key}___', sprite-svg-uri(quote($value)));
       │                                                                                  ^^^^^^^^^^^^^
       ╵
      src/scss/_sprites.scss 65:82  sprite()
      stdin 4:5                     root stylesheet
          in /home/cstornowski/Downloads/svg-spritemap-webpack-plugin/examples/variables/src/scss/_sprites.scss (line 65, column 82)
```

### git diff:

```bash
diff --git a/examples/variables/webpack.config.js b/examples/variables/webpack.config.js
index 9f6c651..7ef8832 100644
--- a/examples/variables/webpack.config.js
+++ b/examples/variables/webpack.config.js
@@ -1,6 +1,8 @@
 const path = require('path');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const SVGSpritemapPlugin = require('../../lib');
+const sass = require('sass');
+const fibers = require('fibers');
 
 module.exports = {
     module: {
@@ -8,8 +10,11 @@ module.exports = {
             test: /\.scss$/,
             use: [
                 MiniCssExtractPlugin.loader,
-                'css-loader',
-                'sass-loader'
+                   {loader:'css-loader'},
+                   {loader: 'sass-loader', options: {
+                                implementation: sass,
+                                fiber: fibers
+                            }}
             ]
         }]
     },
```

### node packages installation:

`npm install mini-css-extract-plugin css-loader style-loader sass-loader sass fibers less-loader less html-webpack-plugin webpack-dev-server --no-save`

**Comment by cascornelissen** [comment]
Thanks for the issue, I'm quite busy the coming few weeks... Are you able to debug this a little by putting some [`@debug` directives](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#debug) with multiple values and figuring out which function expects a string and why this isn't a string?

https://github.com/cascornelissen/svg-spritemap-webpack-plugin/blob/02a700b6240eb7c9a43710aaf186de514e42bfd2/lib/templates/styles.scss#L58

**Comment by ChristianStornowski** [comment]
I only found a documentation from jetbrains `sass_functions.scss`

```scss
/**
 * Add quotes to a string if the string isn't quoted,
 * or returns the same string if it is.
 * 
 * @see #unquote
 * @example
 * quote("foo") => "foo"
 * quote(foo) => "foo"
 * @overload quote($string)
 * @param $string [Sass::Script::Value::String]
 * @return [Sass::Script::Value::String]
 * @raise [ArgumentError] if `$string` isn't a string
 */
@function quote($string) { /* stub */ }
```

`quote` needs a string. node-sass has no problem with your plugin but dart-sass excepts a string.

For now I don't have more information.

**Comment by cascornelissen** [comment]
Fixed in #66
